app/scripts/controllers/alert-controller.test.ts

```diff
@@ -28,230 +28,196 @@ const EMPTY_ACCOUNT = {
+        controller.setWeb3ShimUsageRecorded('testWeb3ShimUsageOrigin');
+        expect(controller.state.web3ShimUsageOrigins).toStrictEqual({
+          testWeb3ShimUsageOrigin: 1,
+        });
+        expect(
+          messenger.call('AlertController:getState').web3ShimUsageOrigins,
+        ).toStrictEqual({
+          testWeb3ShimUsageOrigin: 1,
+        });
       });
     });
-    it('should set origin of web3ShimUsageOrigins to dismissed', () => {
-      alertController.setWeb3ShimUsageAlertDismissed('testWeb3ShimUsageOrigin');
-      expect(
-        alertController.store.getState().web3ShimUsageOrigins,
-      ).toStrictEqual({
-        testWeb3ShimUsageOrigin: 2,
-      });
-      expect(
-        controllerMessenger.call('AlertController:getState')
-          .web3ShimUsageOrigins,
-      ).toStrictEqual({
-        testWeb3ShimUsageOrigin: 2,
+    it('should set origin of web3ShimUsageOrigins to dismissed', async () => {
+      await withController(({ controller, messenger }) => {
+        controller.setWeb3ShimUsageAlertDismissed('testWeb3ShimUsageOrigin');
+        expect(controller.state.web3ShimUsageOrigins).toStrictEqual({
+          testWeb3ShimUsageOrigin: 2,
+        });
+        expect(
+          messenger.call('AlertController:getState').web3ShimUsageOrigins,
+        ).toStrictEqual({
+          testWeb3ShimUsageOrigin: 2,
+        });
       });
     });
   });
 
   describe('selectedAccount change', () => {
-    it('should set unconnectedAccountAlertShownOrigins to {}', () => {
-      controllerMessenger.publish('AccountsController:selectedAccountChange', {
-        id: '',
-        address: '0x1234567',
-        options: {},
-        methods: [],
-        type: 'eip155:eoa',
-        metadata: {
-          name: '',
-          keyring: {
-            type: '',
+    it('should set unconnectedAccountAlertShownOrigins to {}', async () => {
+      await withController(({ controller, messenger }) => {
+        messenger.publish('AccountsController:selectedAccountChange', {
+          id: '',
+          address: '0x1234567',
+          options: {},
+          methods: [],
+          type: 'eip155:eoa',
+          metadata: {
+            name: '',
+            keyring: {
+              type: '',
+            },
+            importTime: 0,
           },
-          importTime: 0,
-        },
+        });
+        expect(
+          controller.state.unconnectedAccountAlertShownOrigins,
+        ).toStrictEqual({});
+        expect(
+          messenger.call('AlertController:getState')
+            .unconnectedAccountAlertShownOrigins,
+        ).toStrictEqual({});
       });
-      expect(
-        alertController.store.getState().unconnectedAccountAlertShownOrigins,
-      ).toStrictEqual({});
-      expect(
-        controllerMessenger.call('AlertController:getState')
-          .unconnectedAccountAlertShownOrigins,
-      ).toStrictEqual({});
     });
   });
 
   describe('AlertController:getState', () => {
```

Comment:
It looks like the AlertController had tests for `getState` as well. Do we need these anymore either? Similar to the `stateChange` event, this should already be tested by BaseController.